src/test/archive/archive_node_tests/archive_node_tests.ml

```diff
@@ -27,6 +27,70 @@ let assert_archived_blocks ~archive_uri ~expected =
       actual_blocks_count expected ()
   else Deferred.unit
 
+(* Convert performance metrics to a JSON format suitable for output *)
+(* The metrics are expected to be a list of tuples (operation, avg_time) *)
+(* where operation is a string and avg_time is a float representing the average time in milliseconds *)
+let perf_metrics_to_yojson metrics =
+  let json_list =
+    List.map metrics ~f:(fun (operation, avg_time) ->
+        `Assoc
+          [ ("operation", `String operation); ("avg_time_ms", `Float avg_time) ] )
+  in
+  `List json_list
+
+(* Extract performance metrics from the log file *)
+(* where X is a floating point number representing the time taken for the operation *)
+(* log output should be in JSON format *)
+(* Example log line: {..., "message": "Operation took 123.45 ms"} *)
+(* The function will return a map of operation names to their average time in milliseconds *)
+let extract_perf_metrics log_file =
+  let open Deferred.Let_syntax in
+  let%bind lines = Reader.file_lines log_file in
+  let perf_metrics =
+    List.filter_map lines ~f:(fun line ->
+        if String.is_empty line then None
+        else
+          match Logger.Message.of_yojson (Yojson.Safe.from_string line) with
+          | Ok entry ->
+              if String.Map.mem entry.metadata "is_perf_metric" then
+                let time_in_ms =
+                  String.Map.find entry.metadata "elapsed"
+                  |> Option.value_exn
+                       ~message:
+                         ("Missing elapsed in log entry in log line: " ^ line)
+                  |> Yojson.Safe.to_string |> Float.of_string
```

Comment:
It's better to use `Yojson.Safe.Util.to_float` to avoid an additional level of converison.